ui/components/app/confirm-page-container/enableEIP1559V2-notice/enableEIP1559V2-notice.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import Box from '../../../ui/box';
+import Dialog from '../../../ui/dialog';
+import Typography from '../../../ui/typography/typography';
+import {
+  COLORS,
+  TYPOGRAPHY,
+  TEXT_ALIGN,
+  FONT_WEIGHT,
+  DISPLAY,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import Button from '../../../ui/button';
+import { EXPERIMENTAL_ROUTE } from '../../../../helpers/constants/routes';
+import { setEnableEIP1559V2NoticeDismissed } from '../../../../store/actions';
+import { getEnableEIP1559V2NoticeDismissed } from '../../../../ducks/metamask/metamask';
+import { getEIP1559V2Enabled } from '../../../../selectors';
+
+const EIP_1559_V2_ENABLED =
+  // This is a string in unit tests but is a boolean in the browser
+  process.env.EIP_1559_V2 === true || process.env.EIP_1559_V2 === 'true';
+
+export default function EnableEIP1559V2Notice({ isFirstAlert }) {
+  const t = useI18nContext();
+  const history = useHistory();
+  const enableEIP1559V2NoticeDismissed = useSelector(
+    getEnableEIP1559V2NoticeDismissed,
+  );
+  const eip1559V2Enabled = useSelector(getEIP1559V2Enabled);
+
+  if (
+    !EIP_1559_V2_ENABLED ||
+    eip1559V2Enabled ||
+    enableEIP1559V2NoticeDismissed
+  ) {
+    return null;
+  }
+
+  return (
+    <Box margin={[0, 4, 4, 4]} marginTop={isFirstAlert ? 4 : 0}>
+      <Dialog type="message" className="enableEIP1559V2-notice__message">
+        <button
+          onClick={setEnableEIP1559V2NoticeDismissed}
+          className="enableEIP1559V2-notice__message__close-button"
```

Comment:
[REQ] We shouldn't nest blocks like this with our BEM syntax. This should be enableEIP1559v2-notce__close-button, and your style should be one less level indented. If you wanted to only target enableEIP1559v2-notice__message > enableEIP1559vs-notice__close-button then you could do &__message  &__close-button. Let me know if that doesn't make sense and I can try and do a better job of explaining

Reply:
Good point, I just fixed.